c_common/front_end_common_lib/include/spinn_extra.h

```diff
@@ -0,0 +1,1106 @@
+ASSERT_WORD_SIZED(dma_stats_control_t);
+
+// ---------------------------------------------------------------------
+// 8. Communications controller
+
+typedef union {
+    struct {
+        uchar parity : 1;
+        uchar payload : 1;
+        uchar timestamp : 2;
+        uchar : 2;
+        uchar type : 2;
+    };
+    struct {
+        uchar : 4;
+        uchar emergency_routing : 2;
+        uchar : 2;
+    } mc;
+    struct {
+        uchar : 4;
+        uchar seq_code : 2;
+        uchar : 2;
+    } p2p;
+    struct {
+        uchar : 2;
+        uchar route : 3;
+        uchar mem_or_normal : 1;
+        uchar : 2;
+    } nn;
+    struct {
+        uchar : 4;
+        uchar emergency_routing : 2;
+        uchar : 2;
+    } fr;
+    uchar value;
+} spinnaker_packet_control_byte_t;
+
+enum {
+    SPINNAKER_PACKET_TYPE_MC = 0,
+    SPINNAKER_PACKET_TYPE_P2P = 1,
+    SPINNAKER_PACKET_TYPE_NN = 2,
+    SPINNAKER_PACKET_TYPE_FR = 3,
+};
+
+typedef struct {
+    uint : 16;
+    uint control_byte : 8;
+    uint : 4;
+    uint not_full : 1;
+    uint overrun : 1;
+    uint full : 1;
+    uint empty : 1;
+} comms_tx_control_t;
+
+typedef struct {
+    uint multicast : 1;
+    uint point_to_point : 1;
+    uint nearest_neighbour : 1;
+    uint fixed_route : 1;
+    uint : 12;
+    uint control_byte : 8;
+    uint route : 3;
+    uint : 1;
+    uint error_free : 1;
+    uint framing_error : 1;
+    uint parity_error : 1;
+    uint received : 1;
+} comms_rx_status_t;
+
+typedef struct {
+    uint p2p_source_id : 16;
+    uint : 8;
+    uint route : 3;
+    uint : 5;
+} comms_source_addr_t;
+
+typedef struct {
+    comms_tx_control_t tx_control;
+    uint tx_data;
+    uint tx_key;
+    comms_rx_status_t rx_status;
+    const uint rx_data;
+    const uint rx_key;
+    comms_source_addr_t source_addr;
+    const uint _test;
+} comms_ctl_t;
+
+ASSERT_WORD_SIZED(comms_tx_control_t);
+ASSERT_WORD_SIZED(comms_rx_status_t);
+ASSERT_WORD_SIZED(comms_source_addr_t);
+
+// ---------------------------------------------------------------------
+// 9. Communications NoC
+
+// No registers
+
+// ---------------------------------------------------------------------
+// 10. SpiNNaker Router
+
+typedef struct {
+    uint route_packets_enable : 1;
+    uint error_interrupt_enable : 1;
+    uint dump_interrupt_enable : 1;
+    uint count_timestamp_errors : 1;
+    uint count_framing_errors : 1;
+    uint count_parity_errors : 1;
+    uint time_phase : 2;
+    uint monitor_processor : 5;
+    uint : 2;
+    uint reinit_wait_counters : 1;
+    uint begin_emergency_wait_time : 8;
+    uint drop_wait_time : 8;
+} router_control_t;
+
+typedef struct {
+    uint interrupt_active_for_diagnostic_counter : 16;
+    uint busy : 1;
+    uint : 7;
+    uint output_stage : 2;
+    uint : 3;
+    uint interrupt_active_dump : 1;
+    uint interrupt_active_error : 1;
+    uint interrupt_active : 1;
+} router_status_t;
+
+enum output_stage {
+    output_stage_empty,
```

Comment:
cap enum names. for consistency with rest of the tools. (also i was thinking i was looking for a variable for a while when reviewing later files)

Reply:
Capitalised